### PR TITLE
fix dense in Structured Metadata Table

### LINKED ISSUE #1735: Dense property not applying to Structured Metadata Table
Dense property does not change how StructuredMetadataTable is rendered 

## Expected Behavior
Should use dense table when dense prop is true

## Current Behavior
Seemingly there is no change when dense is set to true on StructuredMetdataTable

## Possible Solution

I may have missread this but the change for this issue https://github.com/spotify/backstage/issues/1283 Was to apply the dense property to the Table, but it looks like in `StructuredMetadataTable.tsx` this prop isn't passed into the table. Maybe it's as simple as that to fix this ?

Would be happy to contribute if so ! 😄 

## Steps to Reproduce

Create Structured Metadata table and set Dense to true

## Your Environment

- NodeJS Version (v12): 12
- Operating System and Version (e.g. Ubuntu 14.04): Mac OSX
- Browser Information: Chrome

### PR DESCRIPTION
**Pull request by samiramkr**
#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

This PR closes: https://github.com/spotify/backstage/issues/1735

Screenshot 2020-09-23 at 14 59 33

### CODE REVIEW

File: packages/core/src/components/StructuredMetadataTable/MetadataTable.tsx [hunk 1]
```diff
@@ -71,13 +71,7 @@ export const MetadataTable = ({
   dense?: boolean;
   children: React.ReactNode;
 }) => (
-  <Table>
-    {!dense && (
-      <colgroup>
-        <col style={{ width: 'auto' }} />
-        <col style={{ width: '100%' }} />
-      </colgroup>
-    )}
```
Comment (samiramkr) [inline packages/core/src/components/StructuredMetadataTable/MetadataTable.tsx:80]: It seems these lines does not have any affects.

File: packages/core/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx [hunk 2]
```diff
@@ -142,17 +142,17 @@ const TableItem = ({
...
-    const { metadata, dense, options } = this.props;
-    const metadataItems = mapToItems(metadata, options || {});
+};
 
-    return <MetadataTable dense={dense}>{metadataItems}</MetadataTable>;
-  }
-}
+export const StructuredMetadataTable = ({
+  metadata,
+  dense = true,
```
[the remark this answers is not in the document]
Comment (samiramkr) [reply packages/core/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx:153]: Yes you are right, I thought same you but I found some comments from Stefan which I understood StructuredMetadataTable should be dense by default so I set it true as a default value.

[the remark this answers is not in the document]
Comment (marcuseide) [reply packages/core/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx:153]: @stefanalund opinions?

[the remark this answers is not in the document]
Comment (stefanalund) [reply packages/core/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx:153]: I think this should be dense by default.

**Review by stefanalund — APPROVED** [review]

**Pull request merged**